src/nvc.rs

```diff
@@ -459,7 +573,23 @@ impl Nvc {
                         }
                         OPCODE_SYSTEM_REGISTER_ID_PSW => self.set_reg_psw(value),
                         OPCODE_SYSTEM_REGISTER_ID_CHCW => {
-                            logln!("WARNING: ldsr chcw not yet implemented (value: 0x{:08x})", value);
+                            if self.chcw_cache_enabled != ((value >> 1) & 0x01 == 1) {
+                                logln!("ldsr chcw cache enabled changed from {} to {}", self.chcw_cache_enabled, !self.chcw_cache_enabled);
+                                self.chcw_cache_enabled = !self.chcw_cache_enabled;
+                            }
+
+                            if value & 0x01 == 1 {
+                                let entry_count = ((value >> 8) & 0x7ffff) as usize;
+                                let entry_start = (value >> 20) as usize;
+                                logln!("ldsr chcw request to clear cache for start entry: {}, entry count: {}", entry_start, entry_count);
+                                self.cpu_cache.clear_entries(entry_start, entry_count);
+                            } else if value >> 4 & 0x01 == 1 {
```

Comment:
nit: should probably have parens around the shift here to be extra clear about the precedence (as in `(value >> 4) & 0x01`)